src/expressions/operators/arithmetic/BinaryOperator.ts

```diff
@@ -79,14 +72,68 @@ export function generateBinaryOperatorFunction(
 		let retType = returnTypeMap[hash(ValueType.XSNUMERIC, ValueType.XSNUMERIC, operator)];
 		if (!retType) retType = determineReturnType(typeA, typeB);
 		if (operator === 'divOp' && retType === ValueType.XSINTEGER) retType = ValueType.XSDECIMAL;
-		if (operator === 'idivOp') return iDivOpChecksFunction(applyCastFunctions, fun);
-		return [
-			(a, b) => {
-				const { castA, castB } = applyCastFunctions(a, b);
-				return createAtomicValue(fun(castA.value, castB.value), retType);
-			},
-			retType,
-		];
+		if (operator === 'idivOp') return iDivOpChecksFunction(applyCastFunctions, fun)[0];
```

Comment:
why is there a `[0]` added at the end of line 75?

Reply:
That's because of the change in return type.